### PR TITLE
docs: Update Document organization structure

### LINKED ISSUE #192: e2e-framework project documentation site
Contributors of this project work hard to provide documentation with new functionalities (see the ./example dir). However, the project does not have a centralized location that gathers and curates the documentation for newcomers.  

As a user of this project, I would like a single location where feature documentation is curated and organized into a subdirectory or a separated project site (or both). That way newcomers can quickly find answers and this would help with adoption.

### PR DESCRIPTION
**Pull request by harshanarayana**
Closes #192 

### Discussion Items
- [ ] How do we re structure README from inside the examples ?

### CODE REVIEW

**Comment by harshanarayana** [comment]
@vladimirvivien  to help discuss the re-grouping of Examples document, I have done it for two files. Wonder what would be the best way to org them

**Review by vladimirvivien — COMMENTED** [review]
@harshanarayana apologies for the late reply.
I think you are on the right track.
I don't think you need to move the markdown docs from the `examples` directory.
For now, you can just copy them into the `docs` directory as to keep the examples complete.

**Comment by harshanarayana** [comment]
> For now, you can just copy them into the docs directory as to keep the examples complete.

You mean keep two copies of it ?

**Comment by vladimirvivien** [comment]
@harshanarayana or keep a README.md in examples that links to where the respective example is documented.  That way, if I browse from example directory I can get back to documentation about example.

**Review by ShwethaKumbla — COMMENTED** [review]
LGTM

**Comment by harshanarayana** [comment]
> That way, if I browse from example directory I can get back to documentation about example.

I thought I can do this easily and created a few symlinks. Seems like github won't redirect you to the symlink location if you click on the file :( It just show the path for which the current file links to.  (The changes are pushed to the branch for now, I will revert them. Leaving them for now just to show you what it looks like.)

Shall I add a section on the `docs/README.md` that links you to different set of Examples ?  cc @vladimirvivien

File: docs/README.md
Comment (vladimirvivien) [inline docs/README.md:1]: @harshanarayana Hi 👋 
Please add a table of content here that links to the different sections. That way, when a user clicks on docks directory, the list of available content will be displayed.

**Review by vladimirvivien — COMMENTED** [review]
Hi @harshanarayana 
Apologies for the late reply.
I have added a comment. I think it's almost there.

File: docs/README.md
Comment (harshanarayana) [reply docs/README.md:1]: Ack. Pushing the changes now.

File: docs/README.md [hunk 1]
```diff
@@ -1,3 +1,61 @@
...
+  - [Leveraging CLI Flags and Adding Custom Flags](#leveraging-cli-flags-and-adding-custom-flags)
+  - [Test Filtering](#test-filtering)
+  - [Using Custom Decoders](#using-custom-decoders)
+  - [Custom Resource Definition](#custom-resource-definition)
+  - [Runtime Features](#runtime-features)
+  - [Multi Cluster Tests](#multi-cluster-tests)
+  - [Third Party Integration](#third-party-integration)
+
+
 # Go E2E Test Framework for Kubernetes
```
Comment (vladimirvivien) [inline docs/README.md:13]: Hi @harshanarayana this line seems to have moved to line 13. I think It is the title and should be at the top, correct?

**Review by vladimirvivien — COMMENTED** [review]
@harshanarayana I have one question. Other than that, I think this is ready to go.

File: docs/README.md (hunk 1 above)
Comment (harshanarayana) [reply docs/README.md:13]: /Ack. You are right. Let me re order that bit

Comment (harshanarayana) [reply docs/README.md]: Done. PTAL when possible @vladimirvivien

**Review by vladimirvivien — APPROVED** [review]
Thanks @harshanarayana 

/lgtm
/approve

**Pull request merged**